src/page/Auth/FindPassword/hooks/useRouteCheck.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+const useRouteCheck = (prevState: string, entryRoute: string) => {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const 이전_상태를_가지고_넘어왔는가 = location.state && (prevState in location.state);
+    if (!이전_상태를_가지고_넘어왔는가) {
+      navigate(entryRoute, { replace: true });
+    }
+  }, [location.state, navigate, entryRoute, prevState]);
+};
+
+export default useRouteCheck;
```

Comment:
1. 일부로 반환되지않는 hook을 유도하신건가요? 아니라면 현재 구현된 hook에서 반환되지 않는 로직을 개선하는 것이 어떨까요? 

개인적인 생각으론 반환이 필요없는 hook 구조로 사용한다면 클라 측면에서 단순화된다는 장점이 있지만, 
반환되는 hook 구조의 경우 향후 `checkRoute` 외에 추가 기능이 필요할 경우, 로직 분리와 유지보수 측면에서도 유리할것같아요. 
사용하는 곳에서도 훨씬 파악하기도 편할것같구요. (제가 생각한 장단점은 여기까지입니다..)
ex. 
```js 
const {checkRoute : checkPasswordRoute} = useRouteCheck('authCheck', '/find-password');
```


2. `이전_상태를_가지고_넘어왔는가` 는 적당한 이름으로 대체할 필요가 있을것같아요ㅎㅎ

3. `prevState` 보다는 route 관련한 hook이니까 `prevRoute` 정도도 괜찮을것같아요

```suggestion
const useRouteCheck = (prevState: string, entryRoute: string) => {
  const location = useLocation();
  const navigate = useNavigate();

const checkRoute = () => {
  useEffect(() => {
    const 이전_상태를_가지고_넘어왔는가 = location.state && (prevState in location.state);
    if (!이전_상태를_가지고_넘어왔는가) {
      navigate(entryRoute, { replace: true });
    }
  }, [location.state, navigate, entryRoute, prevState]);
};
};

export default {checkRoute};
```

Reply:
1. 현재 `useRouteCheck`가 한가지의 일만 하고 있는데, 미래의 확장성까지 고려해서 `useRouteCheck` 내부에 `checkRoute`기능으로 분리하고 추후 다른 기능이 추가할 수 있도록 하자는 의도로 해석되는데 맞을까요?
1-1. 만약 `useRouteCheck`에 다른 기능이 추가된다면 예를 들어서 어떤 기능이 추가될 수 있을까요?
1-2. 만약 다른 기능이 추가된다고 하면 `useRouteCheck`내부에 여러가지 기능을 만드는 것과 각 기능별로 hook을 분리하는 것 중 어떤게 더 나을지 궁금합니다!
2. 한글변수는 원빈이의 강력한 추천으로 사용해보았습니다만... 사실 이정도 변수는 `isPrevState`나 `hasPrevState`정도로도 사용할 수 있긴 하겠네요😅 앞으로도 한글변수를 적극적으로 사용해볼려고 하는데 네이밍컨벤션에 위배될까요? 조금 복잡한 변수명은 한글변수가 훨씬 가독성 있다고 느껴집니다..!
3. `prevRoute`가 더 적절해보이네요..! 수정했습니다! acc88a7576113b27b7d449eb3ada168a2e72f8b2

Reply:
1. 네 맞습니다. 
- 추가될 기능이라고 하면 페이지 유효성 체크(페이지가 존재하지않는다면 404페이지로 리다이렉트), 특정 권한 검사, 특정한 url에 대한 매개변수 처리 등이 있을것같아요. 사실 모든 기능이 지금 꼭 필요하진않으니,  만약 필요하다면 그때 적절히 조율하는것도 좋겠네요.
- 처음엔 `useRoute` hook정도로 더 추상화해서 route 와 관련된 모든 기능을 추가하는 방식을 생각했었는데,  너무 응집되면 복잡해질수도있을것같아서, 그냥 지금처럼 적당한 기능별로 나누는것도 괜찮을것같네요ㅎㅎ 

+함수(또는 hook)의 이름도 `validatePreviousRoute` 등으로 이전  route를 검사한다는 느낌을 주면 좋을것같아요.
또한 이전 경로를 가지고있지 않을때 적절한 에러를 반환해주는 에러 처리도 필요할것같아요.  

2. 조금 더 의미 전달을 위해 `이전_경로를_가지고_있는가` 정도도 괜찮을것같네요. 프로젝트에 한글변수를 적용한다면, 어디에 어떻게 적용할지 적절한 기준이 필요할거같아요.